game/scripts/vscripts/Sections/Chapter2/SectionCreeps.ts

```diff
@@ -0,0 +1,337 @@
+                    tg.textDialog(LocalizationKey.Script_2_Creeps_14, context => context[CustomNpcKeys.SunsFanMudGolem], 4),
+                    tg.immediate(context => {
+                        goalKillSniper.start()
+                        const sniper: CDOTA_BaseNPC = context[chapter2SpecificKeys.sniperEnemyHero]
+                        if (sniper) {
+                            if (sniper.HasModifier(modifier_sniper_deny_chapter2_creeps.name)) {
+                                const modifier = sniper.FindModifierByName(modifier_sniper_deny_chapter2_creeps.name) as modifier_sniper_deny_chapter2_creeps;
+                                if (modifier) {
+                                    modifier.sniperDenyingOwnCreeps = false;
+                                }
+                            }
+                        }
+                    }),
+                    tg.completeOnCheck(context => {
+                        const sniper = context[chapter2SpecificKeys.sniperEnemyHero] as CDOTA_BaseNPC;
+                        return !sniper.IsAlive()
+                    }, 0.5),
+                    tg.playGlobalSound("SniperDiesLamely.mp3", true),
+                    tg.immediate(() => {
+                        goalKillSniper.complete()
+
+                        if (lastHitTimer) Timers.RemoveTimer(lastHitTimer)
+                    })
+                ])
+            ])
+        ]))
+
+    graph.start(GameRules.Addon.context, () => {
+        print("Completed", sectionName)
+        complete()
+    })
+}
+
+const onStop = () => {
+    print("Stopping", sectionName);
+
+    const hero = getPlayerHero();
+    if (hero) {
+        if (hero.HasModifier(modifier_dk_last_hit_chapter2_creeps.name)) {
+            hero.RemoveModifierByName(modifier_dk_last_hit_chapter2_creeps.name);
+        }
+    }
+
+    if (lastHitTimer) {
+        Timers.RemoveTimer(lastHitTimer)
+        lastHitTimer = undefined;
+    }
+
+    const context = GameRules.Addon.context;
+    const radiantCreeps = context[chapter2SpecificKeys.RadiantCreeps]
+    if (radiantCreeps) {
+        if (Array.isArray(radiantCreeps)) {
+            for (const radiantCreep of radiantCreeps) {
+                if (IsValidEntity(radiantCreep)) {
+                    UTIL_Remove(radiantCreep)
+                }
+            }
+        }
+    }
+
+    const direCreeps = context[chapter2SpecificKeys.DireCreeps]
+    if (direCreeps) {
+        if (Array.isArray(direCreeps)) {
+            for (const direCreep of direCreeps) {
+                if (IsValidEntity(direCreep)) {
+                    UTIL_Remove(direCreep)
+                }
+            }
+        }
+    }
+
+    const sniper = context[chapter2SpecificKeys.sniperEnemyHero]
+    if (sniper) {
+        if (IsValidEntity(sniper)) {
+            UTIL_Remove(sniper);
+        }
+        context[chapter2SpecificKeys.sniperEnemyHero] = undefined;
```

Comment:
Suggestion for utils:
```ts
function removeContextEntityIfExists(context, entityKey) {
    const entity = context[entityKey];
    if (entity && IsValidEntity(entity)) {
        UTIL_Remove(entity);
    }
    context[entityKey] = undefined;
}
```

Reply:
Added this utility plus a check for arrays to take care of all instances in an array. Short testing seems to work as intended.